Makefile

```diff
@@ -3,13 +3,49 @@
 SOURCES = AddressResolver.cpp Profile.cpp
 HEADERS = AddressResolver.h Profile.h
 
+KCACHEGRIND = kcachegrind
+
+CFLAGS = -Wall -Wextra -Werror -g -fno-omit-frame-pointer
+
 all: pgcollect pginfo pgconvert
 
 pgcollect: pgcollect.c
-	$(CC) -std=gnu99 -Wall -Wextra -g -D_GNU_SOURCE -o pgcollect pgcollect.c ${FLAGS}
+	$(CC) -std=gnu99  $(CFLAGS) -D_GNU_SOURCE -o pgcollect pgcollect.c   ${FLAGS}
 
 pginfo: pginfo.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pginfo pginfo.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+	$(CXX) -std=c++03 $(CFLAGS)               -o pginfo    pginfo.cpp    $(SOURCES) -ldw -lelf ${FLAGS}
 
 pgconvert: pgconvert.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pgconvert pgconvert.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+	$(CXX) -std=c++03 $(CFLAGS)               -o pgconvert pgconvert.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+
+.PHONY: clean clean-dev clean-check
+
+clean: clean-dev clean-check
+	rm -rf pgcollect pginfo pgconvert
+
+clean-dev:
+	rm -rf *.o
+
+clean-check:
+	rm -rf *check*
+
+callgrind.out.check_ls.pgdata callgrind.out.check.pgdata:
+	$(MAKE) check
+
+check:	pgcollect pgconvert pginfo
+	@echo ""; echo "collecting some data of ls binary (likely without full symbols)..."
+	./pgcollect check_ls.pgdata -F 8192 -s -- ls -l /usr/bin  1>/dev/null
+	@echo ""; echo "collecting data of checking that (guaranteed to have symbols for binary pginfo) ..."
+	./pgcollect check.pgdata -F 16392 -s -- ./pginfo callgraph check_ls.pgdata
+	@echo ""; echo "converting both collections to callgrind format ..."
+	./pgconvert check_ls.pgdata -d object       1> callgrind.out.check_ls.pgdata
```

Comment:
Can we use .callgrind or .grind extension for converted data?

Reply:
kcachegrind searches for callgrind.out.* si that's what I would suggest people to use.
But other than the default in its file open dialog and _potentially_ auto-loading one can use any extension.

It is your project so make an educated decision and this PR will follow that.